mtp_noms_ops/apps/security/forms/check.py

```diff
@@ -172,3 +172,54 @@ def accept_or_reject(self):
             logger.exception(f'Check {self.object_id} could not be actioned')
             self.add_error(None, _('There was an error with your request.'))
             return False
+
+
+class AssignCheckToUserForm(GARequestErrorReportingMixin, forms.Form):
+    assignment = forms.CharField(max_length=10)
+
+    def __init__(self, object_id, request, **kwargs):
+        super().__init__(**kwargs)
+        self.object_id = object_id
+        self.request = request
+
+    @cached_property
+    def session(self):
+        return get_api_session(self.request)
+
+    def get_object(self):
+        """
+        Gets the security detail object, a sender or prisoner profile
+        :return: dict or None if not found
+        """
+        try:
+            return self.session.get(self.get_endpoint_path()).json()
+        except RequestException:
+            self.add_error(None, _('This service is currently unavailable'))
+            return {}
+
+    def get_endpoint_path(self):
+        return f'/security/checks/{self.object_id}/'
+
+    def assign_or_unassign(self):
+        endpoint = self.get_endpoint_path()
+
+        if self.cleaned_data.get('assignment') == 'assign':
```

Comment:
validation ought to really be done by the form automatically before one tries to use it to action something.

Reply:
This isn't validating, it's using the values to determine what's being sent to the API

Reply:
if the form validates what can go in the "assignment" field, you don't need the "else add error" part at the end

Reply:
Cool, think we've got that sorted too